library-backend/library-app/src/test/java/org/internship/library/app/controller/BookControllerIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+package org.internship.library.app.controller;
+
+import org.internship.library.api.Book;
+import org.internship.library.app.LibraryAppConfigTest;
+import org.internship.library.app.persistence.entity.BookEntity;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Integration Testing Class from Controller to DB
+ */
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {LibraryAppConfigTest.class})
+public class BookControllerIntegrationTest {
+
+    TestRestTemplate testRestTemplate = new TestRestTemplate();
+    HttpHeaders headers = new HttpHeaders();
+
+    /**
+     * Verify if get request return a book from database
+     */
+    @Test
+    public void getBook() {
+
+        String uri = "http://localhost:7777/library/book/{id}";
```

Comment:
Define the URL for testing in the application.yaml used for tests. 

These strings should be parameters and should not be defined within the test and duplicated in each method.

Similar to the rest client that you've built.

Reply:
Same for all of the methods below.